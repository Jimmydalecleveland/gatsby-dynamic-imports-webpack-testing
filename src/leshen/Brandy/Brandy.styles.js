import styled from '@emotion/styled'

import Typography from '../Typography'

// This is being exported as a styled component so that
// it can be targeted by other styled components (eg. Stack)
export const Brandy = styled(Typography)``
