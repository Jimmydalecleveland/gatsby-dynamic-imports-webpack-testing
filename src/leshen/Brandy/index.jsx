import React from 'react'
import PropTypes from 'prop-types'

import * as Styled from './Brandy.styles'

const Brandy = ({ symbol, text, variant }) => (
  <Styled.Brandy
    variant={variant}
    dangerouslySetInnerHTML={{ __html: `${symbol || ''}${text}`.trim() }}
  />
)

Brandy.defaultProps = {
  symbol: '',
  variant: 'legal',
}

Brandy.propTypes = {
  symbol: PropTypes.string,
  text: PropTypes.string.isRequired,
  variant: PropTypes.oneOf([
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'legal',
    'feature',
    'body',
    'small',
  ]),
}

export default Brandy
