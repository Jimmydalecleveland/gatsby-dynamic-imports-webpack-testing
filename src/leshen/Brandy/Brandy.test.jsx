import React from 'react'
import { render } from 'setup-tests'
import Brandy from '.'

describe('Brandy', () => {
  test('should render as default', () => {
    const { container } = render(<Brandy variant="legal" text="example" />)
    expect(container).toMatchSnapshot()
  })

  test('should render the symbol if it is provided', () => {
    const { container } = render(
      <Brandy variant="small" text="example" symbol="asdf" />
    )
    expect(container).toMatchSnapshot()
  })

  test('should render only the text if a symbol is not provided', () => {
    const { getByText } = render(<Brandy variant="feature" text="example" />)
    getByText('example')
  })

  test('should render pass the variant prop to the Typography component', () => {
    const { container } = render(
      <>
        <Brandy variant="h1" text="example" />
        <Brandy variant="h2" text="example" />
        <Brandy variant="h3" text="example" />
        <Brandy variant="h4" text="example" />
        <Brandy variant="h5" text="example" />
        <Brandy variant="h6" text="example" />
        <Brandy variant="feature" text="example" />
        <Brandy variant="body" text="example" />
        <Brandy variant="small" text="example" />
        <Brandy variant="legal" text="example" />
      </>
    )

    expect(container).toMatchSnapshot()
  })
})
