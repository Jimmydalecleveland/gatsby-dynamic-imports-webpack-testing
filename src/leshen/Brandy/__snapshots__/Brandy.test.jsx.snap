// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brandy should render as default 1`] = `
.emotion-1 {
  font-size: 0.625rem;
  line-height: 0.75rem;
  opacity: 0.8;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-1 {
    width: 100%;
  }
}

<div>
  <p
    class="emotion-0 emotion-1 emotion-2"
  >
    example
  </p>
</div>
`;

exports[`Brandy should render pass the variant prop to the Typography component 1`] = `
.emotion-28 {
  font-size: 0.625rem;
  line-height: 0.75rem;
  opacity: 0.8;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-28 {
    width: 100%;
  }
}

.emotion-25 {
  font-size: 0.875rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-25 {
    width: 100%;
  }
}

.emotion-19 {
  font-size: 1.125rem;
  line-height: 1.5rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-19 {
    width: 100%;
  }
}

.emotion-1 {
  font-weight: 800;
  font-size: calc(2.7777777777777777vw + 18.666666666666668px);
  line-height: calc(3.0555555555555554vw + 23.333333333333332px);
}

@media (max-width:480px) {
  .emotion-1 {
    font-size: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    font-size: 52px;
  }
}

@media (max-width:480px) {
  .emotion-1 {
    line-height: 38px;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    line-height: 60px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-1 {
    width: 100%;
  }
}

.emotion-4 {
  font-weight: 800;
  font-size: calc(2.7777777777777777vw + 14.666666666666666px);
  line-height: calc(3.3333333333333335vw + 16px);
}

@media (max-width:480px) {
  .emotion-4 {
    font-size: 28px;
  }
}

@media (min-width:1200px) {
  .emotion-4 {
    font-size: 48px;
  }
}

@media (max-width:480px) {
  .emotion-4 {
    line-height: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-4 {
    line-height: 56px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-4 {
    width: 100%;
  }
}

.emotion-7 {
  font-weight: 800;
  font-size: calc(2.2222222222222223vw + 13.333333333333334px);
  line-height: calc(2.7777777777777777vw + 14.666666666666666px);
}

@media (max-width:480px) {
  .emotion-7 {
    font-size: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-7 {
    font-size: 40px;
  }
}

@media (max-width:480px) {
  .emotion-7 {
    line-height: 28px;
  }
}

@media (min-width:1200px) {
  .emotion-7 {
    line-height: 48px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-7 {
    width: 100%;
  }
}

.emotion-10 {
  font-weight: 500;
  font-size: calc(0.8333333333333334vw + 18px);
  line-height: calc(1.6666666666666667vw + 16px);
}

@media (max-width:480px) {
  .emotion-10 {
    font-size: 22px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    font-size: 28px;
  }
}

@media (max-width:480px) {
  .emotion-10 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    line-height: 36px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-10 {
    width: 100%;
  }
}

.emotion-13 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-13 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-13 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-13 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-13 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-13 {
    width: 100%;
  }
}

.emotion-16 {
  font-weight: 500;
  font-size: calc(0.2777777777777778vw + 16.666666666666668px);
  line-height: calc(0.5555555555555556vw + 21.333333333333332px);
}

@media (max-width:480px) {
  .emotion-16 {
    font-size: 18px;
  }
}

@media (min-width:1200px) {
  .emotion-16 {
    font-size: 20px;
  }
}

@media (max-width:480px) {
  .emotion-16 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-16 {
    line-height: 28px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-16 {
    width: 100%;
  }
}

.emotion-22 {
  font-size: 1rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-22 {
    width: 100%;
  }
}

<div>
  <h1
    class="emotion-0 emotion-1 emotion-2"
  >
    example
  </h1>
  <h2
    class="emotion-0 emotion-4 emotion-2"
  >
    example
  </h2>
  <h3
    class="emotion-0 emotion-7 emotion-2"
  >
    example
  </h3>
  <h4
    class="emotion-0 emotion-10 emotion-2"
  >
    example
  </h4>
  <h5
    class="emotion-0 emotion-13 emotion-2"
  >
    example
  </h5>
  <h6
    class="emotion-0 emotion-16 emotion-2"
  >
    example
  </h6>
  <p
    class="emotion-0 emotion-19 emotion-20"
  >
    example
  </p>
  <p
    class="emotion-0 emotion-22 emotion-20"
  >
    example
  </p>
  <p
    class="emotion-0 emotion-25 emotion-20"
  >
    example
  </p>
  <p
    class="emotion-0 emotion-28 emotion-20"
  >
    example
  </p>
</div>
`;

exports[`Brandy should render the symbol if it is provided 1`] = `
.emotion-1 {
  font-size: 0.875rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-1 {
    width: 100%;
  }
}

<div>
  <p
    class="emotion-0 emotion-1 emotion-2"
  >
    asdfexample
  </p>
</div>
`;
